aodntools/ncwriter/template.py

```diff
@@ -308,6 +313,11 @@ def create_variables(self, **kwargs):
             if varattr['_data'] is not None:
                 ncvar[:] = varattr['_data']
 
+            # convert some variables attribute to variable datatype
+            for varattr_to_convert in varattrs_to_convert_to_datatype:
+                if varattr_to_convert in varattr.keys():
+                    varattr[varattr_to_convert] = np.array(varattr[varattr_to_convert]).astype(varattr['_datatype']).item()
```

Comment:
This looks more complicated than it needs to be, and doesn't work for array-valued attributes. This seems to work for me:
```suggestion
                    varattr[varattr_to_convert] = np.array(varattr[varattr_to_convert], dtype=varattr['_datatype'])
```